### DISCUSSION TITLE
Tests must be triggered automatically after an application is deployed

### DISCUSSION THREAD

**Discussion by yasinitskyi** [cypress-io/cypress#16718] [CI setup]
My team have an application which lives in GitLab and deploys from there to Google Cloud once pipeline has successfully finished. The team do not want tests to be part of the pipeline for certain reasons, so they suggest me to set them somewhere else and trigger them automatically after the application is deployed. Currently, the pipeline has the last stage which is DEPLOY, and my test should react after that stage is done and trigger themselves 24/7. Any idea how can I do that?

**Comment by sandra-ouadghiri** [comment]
Sure. This is not a cypress matter at all. This is a pipeline topic, especially a gitlab-ci matter.

You can ask pipeline A in a step to trigger pipeline B. This is well documented.
https://docs.gitlab.com/ee/ci/triggers/README.html#when-used-with-multi-project-pipelines

**Reply by yasinitskyi** [reply]
> Sure. This is not a cypress matter at all. This is a pipeline topic, especially a gitlab-ci matter.
> 
> You can ask pipeline A in a step to trigger pipeline B. This is well documented.
> https://docs.gitlab.com/ee/ci/triggers/README.html#when-used-with-multi-project-pipelines

Thank you! I didn't specify gitlab, other resources will be welcomed too. But what If I need to run stress testing, have the same pipeline be triggered 20 times at once, like mimic 20 user using website at the same time, gitlab can do that?

**Reply by sandra-ouadghiri** [reply]
Hi @yasinitskyi 
Happy I was of any help.
For stressing your app, I would rather advise you to use a framework dedicated to stress test, such as Gatling, Vegeta, k6 or jmeter. Else you will struggle on several problems down the road.

**Comment by sandra-ouadghiri** [comment]
@yasinitskyi > If this is answered, could you please mark this as replied :) ?
